evm_arithmetization/src/generation/prover_input.rs

```diff
@@ -307,6 +308,15 @@ impl<F: Field> GenerationState<F> {
             ))
         }
     }
+    /// Returns a non-jumpdest proof for the address on the top of the stack. A
+    /// non-jumpdest proof is the clossest address to address on the top of
+    /// the stack.
+    fn run_next_non_jumpdest_proof(&mut self) -> Result<U256, ProgramError> {
+        let code = self.get_current_code()?;
+        let address = u256_to_usize(stack_peek(self, 0)?)?;
```

Comment:
Dumb question: is it always safe to downsize the `U256` to `usize` here? If it is always guaranteed to be safe to do so here, maybe that merits a motivating comment, something like `// memory addresses are guaranteed to be < usize::max so this cast is safe`?

Following the code path here it seems like there's a lot of work being done to get the address. Could this be done faster?

Reply:
> is it always safe to downsize the U256 to usize here? 
This would be the top of the stack when performing a jump/jumpi during the execution of an arbitrary contract. So I think it can be anything (but maybe is already range-checked?)